Assets/MXR.SDK/Runtime/Android/Utils/MXRAndroidUtils.Device.cs

```diff
@@ -224,6 +224,64 @@ public static partial class MXRAndroidUtils {
         public static string PicoUIVersion =>
             IsPicoDevice ? AndroidOSBuild.SafeGetStatic<string>("DISPLAY") : "0.0.0";
 
+        /// <summary>
+        /// The minimum Lenovo firmware version required to support casting functionality.
+        /// </summary>
+        private const int CastingSupportedLenovoFirmware = 1072;
+
+        // Lenovo UI VERSION UTILS
+        /// <summary>
+        /// Returns Lenovo's UI version. returns "0.0.0" if current device is not a Lenovo device
+        /// </summary>
+        public static string LenovoUIVersion =>
+            IsLenovoDevice ? AndroidOSBuild.SafeGetStatic<string>("DISPLAY") : "0.0.0";
+
+        /// <summary>
+        /// Checks if the Lenovo UI version is greater than or equal to the specified major and minor version.
+        /// </summary>
+        /// <param name="majorVersion">The major version to compare against.</param>
+        /// <param name="minorVersion">The minor version to compare against.</param>
+        /// <returns>True if the current Lenovo UI version is greater than or equal to the specified version; otherwise, false.</returns>
+        public static bool LenovoVersionIsGreaterThanOrEqualTo(int majorVersion, int minorVersion) {
+            try {
+
+                string lenovoUIVersion = LenovoUIVersion;
+
+                
+                string[] parts = lenovoUIVersion.Split('_');
+                if (parts.Length < 3) {
+                    return false;
+                }
+
+                string currentVersionPart = parts[2].Substring(1); 
+
+                // Extract the major and minor parts from the current version part
+                if (currentVersionPart.Length < 4) {
+                    return false;
+                }
+
+                int currentMajorVersion = int.Parse(currentVersionPart.Substring(0, 4));
+                int currentMinorVersion = currentVersionPart.Length > 4 ? int.Parse(currentVersionPart.Substring(4)) : 0;
+
+
+                if (currentMajorVersion > majorVersion || (currentMajorVersion == majorVersion && currentMinorVersion >= minorVersion)) {
+                    return true;
+                }
+
+                return false;
+            } catch {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Determines if the Lenovo device supports casting by checking if it is a Lenovo device
+        /// and if the firmware version is greater than or equal to the specified casting-supported version.
+        /// </summary>
+        public static bool LenovoDeviceSupportsCasting => IsLenovoDevice && LenovoVersionIsGreaterThanOrEqualTo(CastingSupportedLenovoFirmware, 0);
```

Comment:
+1 to @adrenak 's comment here. I think we should pull this and the constant CastingSupportedLenovoFirmware into our own repo. This isn't really a public util as of now.

As a result, the version comparison could also be internal.